backend/src/main/java/com/redhat/cloud/notifications/auth/principal/ConsolePrincipal.java

```diff
@@ -0,0 +1,32 @@
+package com.redhat.cloud.notifications.auth.principal;
+
+import java.security.Principal;
+
+public abstract class ConsolePrincipal<T extends ConsoleIdentity> implements Principal {
+    private final String name;
+    private final T identity;
+    private static final ConsolePrincipal<ConsoleIdentity> NO_IDENTITY_PRINCIPAL = new ConsolePrincipal<>() { };
```

Comment:
I wasn't sure how to represent it, this is a singleton instance (need to check if is still being used). 

would it be clear if I create a new class instead of instantiating an anonymous one?

Reply:
That singleton instance is fine, but `ConsolePrincipal` shouldn't be marked as `static` IMO. This is super minor issue though.